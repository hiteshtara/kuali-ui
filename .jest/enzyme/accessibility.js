/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
