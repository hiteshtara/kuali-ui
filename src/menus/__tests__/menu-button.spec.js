/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import { ListItem } from '../../lists'
import Icon from '../../icons'

import { MenuButton } from '..'

const noop = () => {}

describe('MenuButton Component', () => {
  it('should render', () => {
    const icon = <Icon name='settings' />

    const wrapper = shallow(
      <MenuButton
        id='button-menu'
        label='Toggle Open a Menu'
        raised
        buttonChildren={icon}
        variant='torch'
      >
        <ListItem onClick={noop} primaryText='Item One' />
        <ListItem onClick={noop} primaryText='Item Two' />
        <ListItem onClick={noop} primaryText='Item Three' />
        <ListItem onClick={noop} primaryText='Item Four' />
      </MenuButton>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
