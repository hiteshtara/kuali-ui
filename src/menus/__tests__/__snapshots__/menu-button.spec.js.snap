// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton Component should render 1`] = `
<Menu
  anchor={
    Object {
      "x": "inner right",
      "y": "overlap",
    }
  }
  cascadingAnchor={
    Object {
      "x": "right",
      "y": "overlap",
    }
  }
  cascadingZDepth={3}
  fixedTo={[Window]}
  id="button-menu"
  listHeightRestricted={true}
  listZDepth={2}
  onBlur={[Function]}
  onClose={[Function]}
  onKeyDown={[Function]}
  position="tr"
  repositionOnResize={false}
  repositionOnScroll={true}
  simplified={true}
  toggle={
    <withInk(withTooltip(Button))
      aria-controls="button-menu"
      block={false}
      component="button"
      icon={false}
      iconBefore={true}
      id={undefined}
      label="Toggle Open a Menu"
      onClick={[Function]}
      raised={true}
      role="menu"
      size="default"
      type="submit"
      variant="torch"
    >
      <Icon
        aria-hidden={true}
        iconClassName="material-icons"
        name="settings"
        variant="default"
      />
    </withInk(withTooltip(Button))>
  }
  visible={false}
>
  <ListItem
    activeClassName="md-text--theme-primary"
    animateNestedItems={true}
    component="div"
    expanderIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        keyboard_arrow_down
      </FontIcon>
    }
    itemComponent="li"
    key="0"
    onClick={[Function]}
    primaryText="Item One"
    renderChildrenOutside={false}
    tabIndex={-1}
  />
  <ListItem
    activeClassName="md-text--theme-primary"
    animateNestedItems={true}
    component="div"
    expanderIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        keyboard_arrow_down
      </FontIcon>
    }
    itemComponent="li"
    key="1"
    onClick={[Function]}
    primaryText="Item Two"
    renderChildrenOutside={false}
    tabIndex={-1}
  />
  <ListItem
    activeClassName="md-text--theme-primary"
    animateNestedItems={true}
    component="div"
    expanderIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        keyboard_arrow_down
      </FontIcon>
    }
    itemComponent="li"
    key="2"
    onClick={[Function]}
    primaryText="Item Three"
    renderChildrenOutside={false}
    tabIndex={-1}
  />
  <ListItem
    activeClassName="md-text--theme-primary"
    animateNestedItems={true}
    component="div"
    expanderIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        keyboard_arrow_down
      </FontIcon>
    }
    itemComponent="li"
    key="3"
    onClick={[Function]}
    primaryText="Item Four"
    renderChildrenOutside={false}
    tabIndex={-1}
  />
</Menu>
`;
